lychee-lib/src/types/mail.rs

```diff
@@ -0,0 +1,19 @@
+use check_if_email_exists::{CheckEmailOutput, Reachable};
+
+/// A crude way to extract error details from the mail output.
+/// This was added because `CheckEmailOutput` doesn't impl `Display`
+pub(crate) fn error_from_output(o: &CheckEmailOutput) -> String {
+    if let Err(_e) = o.misc.as_ref() {
+        return "Error occurred connecting to this email server via SMTP".to_string();
+    } else if let Err(e) = &o.smtp {
+        return format!("{:?}", e);
+    } else if let Err(e) = &o.mx {
+        return format!("{:?}", e);
+    }
+    match &o.is_reachable {
+            Reachable::Safe => "Safe: The email is safe to send",
+            Reachable::Risky => "Risky: The email address appears to exist, but has quality issues that may result in low engagement or a bounce",
+            Reachable::Invalid => "Invalid: Email doesn't exist or is syntactically incorrect",
+            Reachable::Unknown => "Unknown: We're unable to get a valid response from the recipient's email server."
+        }.to_string()
```

Comment:
doesn't seem like indented correctly

Reply:
that's what cargo fmt gave me 🤷‍♀️

Reply:
It's weird, though: when I change the indentation of the match blog, it doesn't change it. :/

Reply:
you might accidentally find a bug...

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=242e40ce131f604a1b1334b86cbec615

Reply:
I know rustfmt gives up on really long lines. But I don't know it gives up so easily...

Reply:
Here's a minimal code example:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=80660b19853fe958557723f5e00682dd

```
fn main() {
    match () {
    _ => "foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo",
};
}
```

Could be a rustfmt bug.

Reply:
Manually changed the formatting to the way it _should_ look like IMHO.